test/unit/org/apache/cassandra/tools/ClearSnapshotTest.java

```diff
@@ -54,32 +56,43 @@ public static void teardown() throws IOException
     @Test
     public void testClearSnapshot_NoArgs() throws IOException
     {
-        ToolRunner tool = runner.invokeNodetool("clearsnapshot");
-        assertEquals(2, tool.getExitCode());
-        assertTrue("Tool stderr: " +  tool.getStderr(), tool.getStderr().contains("Specify snapshot name or --all"));
+        try (ToolRunner tool = runner.invokeNodetool("clearsnapshot"))
```

Comment:
Can we remove the unneeded close?  Looks like all usage switched to calling close but `ToolRunner` really should not support any existing running process as the API doesn't properly work in this case.

Reply:
Referring back to another previous comment. Let's leave this as it is now if you don't mind until we have a final `ToolRunner` version to stick with?

Reply:
yep, lets wait till we agree on the plan